AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConfigurationTests.swift

```diff
@@ -0,0 +1,150 @@
+        let post = Post(title: "title", content: "content", createdAt: time)
+        let id = post.id
+
+        Amplify.DataStore.save(post, completion: { result in
+            switch result {
+            case .success:
+                print("Post saved successfully!")
+                databaseCreationDone.fulfill()
+            case .failure(let error):
+                print("Error saving post \(error)")
+            }
+        })
+        wait(for: [databaseCreationDone], timeout: 10)
+
+        Amplify.reset()
+
+        let databaseRecreationDone = expectation(description: "Database remains")
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels())
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure()
+            print("Amplify configured with DataStore plugin")
+        } catch {
+            print("Failed to initialize Amplify with \(error)")
+        }
+
+        Amplify.DataStore.query(Post.self, byId: id) { result in
+            switch result {
+            case .success(let postRes):
+                XCTAssertNotNil(postRes)
+                XCTAssertEqual(postRes!.id, id)
+                XCTAssertEqual(postRes!.title, "title")
+                XCTAssertEqual(postRes!.content, "content")
+                XCTAssertEqual(postRes!.createdAt, time)
+                databaseRecreationDone.fulfill()
+            case .failure(let error):
+                print("Error retrieving posts \(error)")
+                XCTFail("Database is not the same")
+            }
+        }
+
+        wait(for: [databaseRecreationDone], timeout: 10)
+
+        Amplify.DataStore.clear(completion: { _ in })
+    }
+
+    func testConfigWithDifferentSchema() throws {
+
+        let databaseCreationDone = expectation(description: "Database created")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels())
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure()
+            print("Amplify configured with DataStore plugin")
+            databaseCreationDone.fulfill()
+        } catch {
+            print("Failed to initialize Amplify with \(error)")
+        }
+
+        wait(for: [databaseCreationDone], timeout: 10)
+
+        let time = Temporal.DateTime.now()
+        let post = Post(title: "title", content: "content", createdAt: time)
+
+        let saveToDatabaseDone = expectation(description: "Save data to Database done")
+        Amplify.DataStore.save(post, completion: { result in
+            switch result {
+            case .success:
+                print("Post saved successfully!")
+                saveToDatabaseDone.fulfill()
+            case .failure(let error):
+                print("Error saving post \(error)")
+            }
+        })
+        wait(for: [saveToDatabaseDone], timeout: 10)
+
+        ModelRegistry.reset()
+        Amplify.reset()
```

Comment:
does `ModelRegistry` not get reset after Amplify reset? Amplify reset calls each plugins reset, DataStore should try to reset the Registry as well then, if it's not

Reply:
Ya, the ModelRegistry.reset() got called inside DataStore.reset(), I'll remove ModelRegistry.reset()